### PR TITLE
Readd functionality from masche842 libui-rs fork

### PR DESCRIPTION
**Pull request by masche842**
Hi Leo,

as bespoken on reddit. Let's do this. 

I've merged my changes onto your libui-rs fork, and readded all the functionality I need at the moment to iui. It works for me on mac OS, but I'm pretty sure there are issues on other platforms. 
Moreover, I'm still using my own libui-fork (git@github.com:masche842/libui) which diverged from libui upstream quite a bit.

Nonetheless, this is all I need at the moment and with these changes in iui, there is no reason for me to maintain my own version of libui-rs. Instead, I can help you with iui.

### CODE REVIEW

**Comment by NoraCodes** [comment]
This is a great PR, thank you. It'll take me a bit to get it into the shape I want (re: documentation especially) but I'll definitely accept it. Thanks!

**Comment by NoraCodes** [comment]
Ok, after going through the PR, I like it as is except that I'm concerned about tracking a non-mainstream branch of libui. Are you planning on contributing your changes upstream?

**Comment by masche842** [comment]
If I checked correctly, your current version is based on https://github.com/arcturu/libui.git
Differences are easy to merge, so as long as arcturu is fine with them, it will be possible.

However I think you mean contributing to andlabs "original" version. I looked into it yesterday, and for a month or so, andlab is pretty actively working on libui again which is great to see!
He also takes stability, multi-platform and backward-compatibility pretty seriously. On one hand this is of course great too, but on the other, he will not easily accept my changes for a variety of reasons.
So, yes, it would be great to see my changes upstream, but I fear it won't happen in the near future. I therefore suggest we stick with arcturu or my branch for the time being and wait for andlabs to get "back on track" first (No easy tasks with 81 open issues and 42 open pull requests). If there is a way to help him with that, let's do that of course.

**Comment by masche842** [comment]
Replace andlab(s) with Pietro, should really have looked up his real name beforehand :-)

**Comment by NoraCodes** [comment]
Makes plenty of sense to me. I'm currently working on some additions in the mas-merge branch, if you'd like to take a look; mostly just refactoring to make it more compatible with the rest of the architecture.

Thanks much for this PR, by the way. Area and the brush functionality was one of the pieces I was most dreading implementing :)

**Comment by masche842** [comment]
mas-merge looks great! Exactly the things that bothered me while moving my code over, but I wanted to check back with you first. And now it's already done :-)
Where can I help you with so we can move forward?
As soon as this is merged I think we can ask pcwalton to put a notice into his readme, so more people get aware of this fork.

**Comment by NoraCodes** [comment]
I'll make a PR into your branch. Then, I'll go over this PR once more and merge it.

**Comment by mitchhentges** [comment]
Hey :wave: I see that an [ill-prepared commit of mine](https://github.com/LeoTindall/libui-rs/pull/4/commits/0f05f6cb84d41ada26ab47d84d8ae1689e60960a) is working its way into forks, which is pretty neat-o!
Note that back when I made that commit/PR I wasn't (and still am not) sure that `Control` _should_ implement `Send`.
There's a small discussion about it in [the original PR](https://github.com/pcwalton/libui-rs/pull/9#issuecomment-233912126).

TL;DR: while you guys are mulling over this PR, please be advised that not all the original PRs (e.g.: mine!) were fully thought out :smile:

**Comment by NoraCodes** [comment]
@mitchhentges Thanks for the the heads up. I'm reverting that commit along with some documentation fixes in commit dee073

**Comment by NoraCodes** [comment]
Closing in favor of #11
